skk-cursor.el

```diff
@@ -42,82 +42,36 @@
-   ;; $B;z$,85$NF~NO%b!<%I$K$F9T$o$l$k$h$&$K(B)$B!#=>$$!"(Bskk-abbrev-mode
-   ;; $B%U%i%0$N%A%'%C%/$NM%@hEY$r>e$2$k!#(B
-   (skk-abbrev-mode
-    skk-cursor-abbrev-color)
-   (skk-jisx0208-latin-mode
-    skk-cursor-jisx0208-latin-color)
-   (skk-katakana
-    skk-cursor-katakana-color)
-   (skk-j-mode
-    skk-cursor-hiragana-color)
-   (skk-jisx0201-mode
-    skk-cursor-jisx0201-color)
-   (t
-    skk-cursor-latin-color)))
+  (cond ((not (and skk-use-color-cursor
+	           skk-mode))
+         (skk-cursor-default-color))
+
+        ;; `skk-start-henkan' $B$NCf$G$O!"(Bskk-j-mode $B%U%i%0$rN)$F$J$,$i!"(B
+        ;; skk-abbrev-mode $B%U%i%0$bN)$F$F$$$k(B ($BJQ498e!"D>8e$KF~NO$9$kJ8(B
+        ;; $B;z$,85$NF~NO%b!<%I$K$F9T$o$l$k$h$&$K(B)$B!#=>$$!"(Bskk-abbrev-mode
+        ;; $B%U%i%0$N%A%'%C%/$NM%@hEY$r>e$2$k!#(B
+        (skk-abbrev-mode skk-cursor-abbrev-color)
+
+        (skk-jisx0208-latin-mode skk-cursor-jisx0208-latin-color)
+        (skk-katakana skk-cursor-katakana-color)
+        (skk-j-mode skk-cursor-hiragana-color)
+        (skk-jisx0201-mode skk-cursor-jisx0201-color)
+        (t skk-cursor-latin-color)))
 
 ;;;###autoload
 (defun skk-cursor-set-1 (color)
-  (cond
-   ((eval-when-compile (featurep 'xemacs))
-    ;;At 10 Jul 2000 16:37:49 +0900,
-    ;;Yoshiki Hayashi <t90553@mail.ecc.u-tokyo.ac.jp> wrote:
-    ;;> foreground $B$r(B background $B$KJQ$($kI,MW$,$"$k$3$H0J30$O!":#$N(B
-    ;;> $B$H$3$m$=$N$^$^$GF0$$$F$$$k$h$&$G$9!#$7$P$i$/(B test $B$7$F$_$^$9!#(B
-    ;;> $B$I$&$b!"(Btext-cursor $B$bIaDL$N(B face $B$N$h$&$G!"(Bforeground $B$,J8(B
-    ;;> $B;z$N?'$r!"(Bbackground $B$,J8;z$NGX7J$N?'$rI=$7$F$$$k$h$&$G$9!#(B
-    (set-face-property 'text-cursor
-		       'background
-		       (or color
-			   (skk-cursor-current-color))
-		       (current-buffer)))
-   (t
-    (when ccc-default-cursor-color
-      (ccc-set-buffer-local-cursor-color
-       (or color
-	   (skk-cursor-current-color)))))))
+  (when ccc-default-cursor-color
+    (ccc-set-buffer-local-cursor-color (or color
+	                                   (skk-cursor-current-color)))))
 
 ;;;###autoload
 (defun skk-cursor-off-1 ()
-  (cond ((eval-when-compile (featurep 'xemacs))
-	 (skk-cursor-set))
-	(t
-	 (when ccc-default-cursor-color
-	   (ccc-set-cursor-color-buffer-local nil)))))
-
-;; advices.
-(when (eval-when-compile (featurep 'xemacs))
-  (defadvice minibuffer-keyboard-quit (before skk-cursor-ad activate)
-    (unless skk-henkan-mode
-      (skk-cursor-set (skk-cursor-default-color)))))
-
-;; Hooks
-(when (eval-when-compile (featurep 'xemacs))
-  (add-hook 'isearch-mode-end-hook #'skk-cursor-set 'append)
-
-  (add-hook 'minibuffer-setup-hook #'skk-cursor-set 'append)
-
-  (add-hook 'minibuffer-exit-hook
-	    (lambda ()
-	      (with-current-buffer (skk-minibuffer-origin)
-		(skk-cursor-set))
-	      (skk-cursor-set skk-cursor-default-color 'force))
-	    'append))
+  (ccc-set-cursor-color-buffer-local nil))
```

Comment:
Hmm, it seems to have lost the condition `(when ccc-default-cursor-color ...)`.
Fixed by https://github.com/skk-dev/ddskk/pull/171